### PR TITLE
Fix ResetTimeout for SubtleAlert

### PR DESCRIPTION
**Pull request by VadymLuchko**
Fixes [AAW-892]

This PR is **[ready]** for review.

### Testing Plan
Manual test

### Summary
This PR provide some changes for existing SubtleAlert code to meet the requirements of ResetTimeout capabilities of SDLCore

### CLA
- [x] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

### CODE REVIEW

File: app/view/sdl/SubtleAlertPopUp.js [hunk 1]
```diff
@@ -253,15 +237,38 @@ SDL.SubtleAlertPopUp = Em.ContainerView.create(
...
+        */
+        setTimerTTS: function(time){
+        var self = SDL.SubtleAlertPopUp;
+        self.set('ttsTimeout', time);
+        clearTimeout(self.ttsTimer);
+        self.ttsTimer = setTimeout(
+          function() {
+            clearTimeout(self.ttsTimer);
+          }, self.ttsTimeout
+        );
```
Comment (vlrm10) [inline app/view/sdl/SubtleAlertPopUp.js]: @VadymLuchko Add indentation.

File: ffw/UIRPC.js [hunk 2]
```diff
@@ -199,8 +199,29 @@ FFW.UI = FFW.RPCObserver.create(
           case 'UI.SubtleAlert':
           {
             if (SDL.SDLModel.onUISubtleAlert(request.params, request.id)) {
-              SDL.SDLController.onSystemContextChange(request.params.appID);
+              if(!request.params.softButtons){
```
Comment (vlrm10) [inline ffw/UIRPC.js]: @VadymLuchko Please convert this expression to boolean, if its array `!!request.params.softButtons.length`

File: ffw/UIRPC.js [hunk 3]
```diff
@@ -199,8 +199,29 @@ FFW.UI = FFW.RPCObserver.create(
...
+                SDL.ResetTimeoutPopUp.extendResetTimeoutRPCs([request.method]);
+                SDL.ResetTimeoutPopUp.extendResetTimeoutCallBack(SDL.SubtleAlertPopUp.setTimerUI , request.method);
+                SDL.ResetTimeoutPopUp.setContext(request.method);
+                SDL.ResetTimeoutPopUp.expandCallbacks(function(){
+                  SDL.SubtleAlertPopUp.deactivate('timeout');
+                }, request.method);
+
+                if (request.params.duration) {
```
Comment (vlrm10) [inline ffw/UIRPC.js]: @VadymLuchko Please use `in` operator to check that the key is exist in the object `'duration' in request.params`

File: ffw/UIRPC.js [hunk 4]
```diff
@@ -199,8 +199,29 @@ FFW.UI = FFW.RPCObserver.create(
...
+                  SDL.ResetTimeoutPopUp.set('timeoutSeconds',
+                    {
+                      'UI.SubtleAlert': request.params.duration/1000,
+                      'TTS.Speak': request.params.duration/1000
```
Comment (vlrm10) [inline ffw/UIRPC.js]: @VadymLuchko duration/1000 - is it convertation to seconds? Move hardcoded number to logical named const

**Review by vlrm10 — CHANGES_REQUESTED** [review]

File: app/view/sdl/SubtleAlertPopUp.js (hunk 1 above)
Comment (VadymLuchko) [reply app/view/sdl/SubtleAlertPopUp.js]: fixed in 9ad1a683d7504e8708d877a776583bfcdf564ce2

File: ffw/UIRPC.js (hunk 2 above)
Comment (VadymLuchko) [reply ffw/UIRPC.js]: fixed in 9ad1a68

File: ffw/UIRPC.js (hunk 3 above)
Comment (VadymLuchko) [reply ffw/UIRPC.js]: fixed in 9ad1a68

File: ffw/UIRPC.js (hunk 4 above)
Comment (VadymLuchko) [reply ffw/UIRPC.js]: fixed in 9ad1a68

**Review by vlrm10 — APPROVED** [review]

**Pull request merged**